### PR TITLE
fix(iam): improve YAML error handling & reporting in IAM ops

### PR DESCRIPTION
**Pull request by metral**
### Proposed changes

fix(iam): improve YAML error handling & reporting in IAM ops
### Related issues (optional)

Fixes https://github.com/pulumi/pulumi-eks/issues/58

### CODE REVIEW

File: nodejs/eks/cluster.ts [hunk 1]
```diff
@@ -314,22 +314,40 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
 
     const roleMappings = pulumi.all([pulumi.output(args.roleMappings || []), instanceRoleMappings])
         .apply(([mappings, instanceMappings]) => {
-            return jsyaml.safeDump([...mappings, ...instanceMappings].map(m => ({
+            mappings.map(m => {
+                if (!checkArnExists(pulumi.output(m.roleArn))) {
+                    throw new Error(`The IAM role ARN ${m.roleArn} could not be properly retrieved or does not exist`);
+                }
+            });
```
Comment (lukehoban) [inline nodejs/eks/cluster.ts]: I don't think this part of the change is necessary - from the original bug report that part of the issue will be fixed more properly by https://github.com/pulumi/pulumi-terraform/issues/262.  We don't in general want to `.get` every resource id the user mentions to us just to verify they didn't make a mistake.

File: nodejs/eks/cluster.ts [hunk 2]
```diff
@@ -314,22 +314,40 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
...
+            const mappingYaml = jsyaml.safeDump([...mappings, ...instanceMappings].map(m => ({
                 rolearn: m.roleArn,
                 username: m.username,
                 groups: m.groups,
             })));
+            if (!mappingYaml) {
+                throw new Error(`The IAM role mappings provided could not be properly serialized to YAML for the aws-auth ConfigMap`);
+            }
```
Comment (lukehoban) [inline nodejs/eks/cluster.ts]: From the linked issue - doesn't `safeDump` throw?  If so, don't we need to catch the internal `sameDump` error and provider a better one?

File: nodejs/eks/cluster.ts (hunk 1 above)
Comment (metral) [reply nodejs/eks/cluster.ts]: Ok

File: nodejs/eks/cluster.ts (hunk 2 above)
Comment (metral) [reply nodejs/eks/cluster.ts]: Done

**Comment by metral** [comment]
Feedback has been addressed. PTAL @lukehoban

**Comment by metral** [comment]
Ping @lukehoban. Any thoughts?

**Comment by metral** [comment]
Going to go ahead and merge to close this out. Feel free to open up an issue if there is a different approach we should take.

**Pull request merged**